github_api/prs.py

```diff
@@ -101,15 +102,25 @@ def close_pr(api, urn, pr):
     return api("patch", path, json=data)
 
 
-def get_pr_last_updated(pr_data):
+def get_pr_last_updated(api, urn, pr_data):
     """ a helper for finding the utc datetime of the last pr branch
     modifications """
-    repo = pr_data["head"]["repo"]
-    if repo:
-        dt = repo["pushed_at"]
-    else:
-        dt = pr_data["created_at"]
-    return arrow.get(dt)
+
+    # read about github's merge test commit
+    # https://developer.github.com/v3/pulls/#get-a-single-pull-request
+    # essentially what github does is, when it sees a new push to a PR, it will
+    # start a background job to test the commit for mergeability.  this
+    # background job creates a magical floating merge commit not really attached
+    # to any branch.  however, this commit is real and its committer is
+    # "GitHub"...and it has a commit date that we can assume is never malicious.
+    # And it's always updated on a new push to the PR's branch
+    #
+    # using all of these facts, we are able to determine the true, reliable,
+    # last update time of the branch backing a PR
+    github_test_merge_commit = pr_data["merge_commit_sha"]
+    commit = commits.get_commit(api, urn, github_test_merge_commit)
```

Comment:
The linked docs don't say so, but is it seems possible that there might not be a `"merge_commit_sha"` before GitHub has tested for mergeability or even while it's doing so. Since you don't account for that, have you tested this?